service/create/service.go

```diff
@@ -315,6 +325,21 @@ func allExistingInstancesMatch(instances *ec2.DescribeInstancesOutput, state EC2
 	return true
 }
 
+func (s *Service) uploadCloudconfigToS3(awsSession *awssession.Session, s3Bucket, path, data string) error {
+	svc := s3.New(awsSession)
```

Comment:
I guess `aws.NewClient()` can return a struct with all the clients we need (ec2, s3, iam).